src/core/ActionMenu/ActionMenu.tsx

```diff
@@ -0,0 +1,240 @@
+import { baseStyles } from './ActionMenu.baseStyles';
+import { ActionMenuItemProps } from './ActionMenuItem/ActionMenuItem';
+import { ActionMenuDividerProps } from './ActionMenuDivider/ActionMenuDivider';
+import { IconOptionsVertical } from 'suomifi-icons';
+
+const baseClassName = 'fi-action-menu';
+export const actionMenuClassNames = {
+  baseClassName,
+  fullWidth: `${baseClassName}--full-width`,
+  button: `${baseClassName}_button`,
+  iconOnly: `${baseClassName}_button--icon-only`,
+};
+
+export interface ActionMenuProps {
+  /** Text content for the button */
+  buttonText?: string;
+  /**
+   * 'default' | 'inverted' | 'secondary' | 'secondaryNoBorder' | 'link'
+   * @default secondary
+   */
+  buttonVariant?: ButtonVariant;
+  /**
+   * Define a label if `buttonText` does not indicate the button purpose,
+   * alternatively you can define aria-labelledby with label-element id
+   */
+  'aria-label'?: string;
+  /** Menu items. Use the `<ActionMenuItem>` or  `<ActionMenuDivider>` components as children */
+  children?:
+    | Array<
+        | React.ReactElement<ActionMenuItemProps>
+        | React.ReactElement<ActionMenuDividerProps>
+      >
+    | React.ReactElement<ActionMenuItemProps>
+    | React.ReactElement<ActionMenuDividerProps>;
+  /** Button container div class name for custom styling */
+  className?: string;
+  /** Menu container div class name for custom styling. Can be used to modify menu "popover" z-index. */
+  menuClassName?: string;
+  /** Disable button usage */
+  disabled?: boolean;
+  /** Ref is forwarded to the button element. Alternative for React `ref` attribute. */
+  forwardedRef?: React.RefObject<HTMLButtonElement>;
+  /** Set components width to 100% */
+  fullWidth?: boolean;
+  /**
+   * Unique id
+   * If no id is specified, one will be generated
+   */
+  id?: string;
+  /** Name used for menu button */
+  name?: string;
+  /** Callback fired on button onBlur
+   * @param {FocusEvent<HTMLButtonElement>} event FocusEvent
+   */
+  onBlur?: (event: FocusEvent<HTMLButtonElement>) => void;
+  /** Callback fired when menu opens */
+  onClose?: () => void;
+  /** Callback fired when menu closes */
+  onOpen?: () => void;
+  /**
+   * Props which are placed at the outermost div of the component.
+   * Can be used, for example, for style
+   */
+  wrapperProps?: Omit<
+    HTMLAttributesIncludingDataAttributes<HTMLDivElement>,
+    'className'
+  >;
+}
+
+const BaseActionMenu = (props: ActionMenuProps) => {
+  const {
+    className,
+    wrapperProps,
+    id,
+    fullWidth,
+    forwardedRef,
+    buttonText,
+    buttonVariant = 'secondary',
+    children,
+    onOpen,
+    onClose,
+    onBlur,
+    menuClassName,
+    ...passProps
+  } = props;
+
+  const openButtonRef = forwardedRef || useRef<HTMLButtonElement>(null);
+  const [menuVisible, setMenuVisible] = useState<boolean>(false);
+  const [ariaExpanded, setAriaExpanded] = useState<boolean>(false);
+  const [selectFirstItem, setSelectFirstItem] =
```

Comment:
The naming of the values is a a bit confusing. Especially `setSelectFirstItem('first');` feels like there's unnecessary repetition. maybe "initialSelectedItem" could be clearer even if the value used for it is pretty much always "first"

Reply:
At first there was only option to set the first item. Later the options were expanded, but the name was forgotten. 
Renamed now.